### DISCUSSION TITLE
How to use python setup.py install or pip install . in pixi?

### DISCUSSION THREAD

**Discussion by chihchiehchen** [prefix-dev/pixi#1521] [Q&A]
Hello, 

Thanks for sharing this great project. I have question on building local repositories in pixi environment. First I used pixi shell to open the virtual environment, then move to the target directory with setup.py in it, finally use pip install .. 

The packages are installed in .pixi/envs/default/lib/python3.11, however, when I tried to import these packages, I always got the error 'No module named xxx'. 

Could you give me some hints. In any case thanks for your patient and help.

**Comment by brian316** [comment]
if you run `which pip` I'm sure you'll see pip is not being used by pixi. use the pyproject.toml to list local path dependencies

**Reply by chihchiehchen** [reply]
Thanks for clarifying. I will try it. Thanks a lot!

**Comment by fr1ll** [comment]
@chihchiehchen you may be able to adapt some guides to migrating from `setup.py` for `uv`:

- [This StackOverflow answer](https://stackoverflow.com/questions/79418598/how-to-do-install-my-custom-package-in-editable-mode-with-uv) has a method using pdm
- [This Github issue](https://github.com/astral-sh/uv/issues/12656) has a much simpler method using only `uv`. I modified it slightly using pixi:

```bash
uv pip compile setup.py -o uv_pip_requirements.txt --no-build-isolation
pixi add --pypi $(awk '!/^((\s)*#)/' uv_pip_requirements.txt)
```
The `awk` bit is to trim comment lines while passing the requirements to `pixi`.

I don't know if the second method will properly handle a complex `setup.py`... I'll find out soon.

**Reply by chihchiehchen** [reply]
Thanks for the information, I'll try it. Thanks a lot.
